### PR TITLE
Added bnvib

### PR DESCRIPTION
**Pull request by steven11sjf**
standard NX library format, already documented online https://switchbrew.org/wiki/BNVIB#Normal_Vibration

### CODE REVIEW

File: src/mercury_engine_data_structures/formats/bnvib.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+from mercury_engine_data_structures.formats.base_resource import BaseResource
+from mercury_engine_data_structures.game_check import Game
+
+
+class VibrationType(Enum):
+    NORMAL = 4
+    LOOP = 12
+    LOOPWAIT = 16
+
+BNVIB = Struct(
```
Comment (Miepee) [inline src/mercury_engine_data_structures/formats/bnvib.py:25]: would be nice to have a link to the switchbrew wiki page here.

File: src/mercury_engine_data_structures/formats/bnvib.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    Int32ul,
+    PrefixedArray,
+    Struct,
+)
+
+from mercury_engine_data_structures.adapters.enum_adapter import EnumAdapter
+from mercury_engine_data_structures.formats.base_resource import BaseResource
+from mercury_engine_data_structures.game_check import Game
+
+# standard switch format. https://switchbrew.org/wiki/BNVIB#Normal_Vibration
```
Comment (Miepee) [inline src/mercury_engine_data_structures/formats/bnvib.py:19]:
```suggestion
# Standard switch format. https://switchbrew.org/wiki/BNVIB#Normal_Vibration
```

**Review by Miepee — APPROVED** [review]

File: src/mercury_engine_data_structures/formats/bnvib.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    LOOPWAIT = 16
+
+BNVIB = Struct(
+    "vibration_type" / EnumAdapter(VibrationType, Int32ul),
+    "_magic" / Const(3, Int16ul),
+    "sample_rate" / Int16ul,
+    "loop_start" / If(construct.this.vibration_type != VibrationType.NORMAL, Int32ul),
+    "loop_end" / If(construct.this.vibration_type != VibrationType.NORMAL, Int32ul),
+    "loop_wait" / If(construct.this.vibration_type == VibrationType.LOOPWAIT, Int32ul),
+    "data" / PrefixedArray(Int32ul, Byte)
```
Comment (duncathan) [inline src/mercury_engine_data_structures/formats/bnvib.py:33]: `Prefixed(Int32ul, GreedyBytes)`

Comment (duncathan) [reply src/mercury_engine_data_structures/formats/bnvib.py:33]: actually looking at the documentation more closely: "The size of a sample is 4 bytes."
this should be `Prefixed(Int32ul, GreedyRange(Int32ul))`

**Pull request merged**